### ISSUE TITLE
In IE11, repeat polyfill seems to fail when uncaught exception is thrown

### ISSUE THREAD

**Issue by chughes87** [zeit/next.js#11355]
# Bug report

## Describe the bug

When testing in IE, I get a white screen when an uncaught exception is thrown rather than the standard error message page or something more user friendly.

## To Reproduce

1. yarn create next-app
2. Add a `throw new Error('boom')` to the top of pages/index.js
3. `yarn dev`
4. Observe the difference between IE11 and other browsers

## Expected behavior

IE11 should have the same behavior for this case as other browsers

## Screenshots

<img width="1026" alt="Screen Shot 2020-03-25 at 4 47 29 PM" src="https://user-images.githubusercontent.com/3608619/77596135-61ad3300-6eb8-11ea-8c72-d2b04f81c535.png">

## System information

- OS: Windows 81
- Browser IE11
- Version of Next.js: 9.3.1

**Comment by timneutkens** [comment]
Feel free to investigate and send a PR

**Comment by jamesmosier** [comment]
I tested with:

* Windows 8.1
* Internet Explorer 11
* Next.js version 9.3.1 (and 9.3.4)

and was not able to reproduce:

![Screen Shot 2020-04-02 at 9 13 12 AM](https://user-images.githubusercontent.com/2854919/78253377-4d71c300-74c2-11ea-8d34-ebeeb1694d67.png)


I started with a fresh Next project via `yarn create next-app --example hello-world hello-world-app`

**Comment by chughes87** [comment]
There were other issues I was seeing in my VM version of IE11 that I didn't see on my actual Windows box. Haven't checked this but it might also be similar. Given the variable results and seeing that other people aren't having this problem I'll just close this.

**Comment by elliottsj** [comment]
I'm facing this issue as well, but it occurs only after attempting to import [nanoid](https://github.com/ai/nanoid)@3.1.3, along with an attempt to transpile the library:

```js
// next.config.js

module.exports = {
  webpack(config, options) {
    config.module.rules.push({
      test: /\.(js|jsx)$/,
      include: [
        // https://github.com/ai/nanoid/releases/tag/3.0.0
        // Must transpile nanoid for IE11 compat
        /nanoid(?!.*node_modules)/,
      ],
      use: options.defaultLoaders.babel,
    });

    return config;
  },
};
```

Full example: https://github.com/Yolk-HQ/next-issue-11355

<img width="1052" alt="Annotation 2020-04-22 121820" src="https://user-images.githubusercontent.com/2177366/80007756-b8486580-8494-11ea-8dd0-c842336d4311.png">

**Comment by jamesmosier** [comment]
Tim would be able to confirm, but it doesn't appear that [Next polyfills `Object.repeat`](https://github.com/zeit/next.js/blob/canary/packages/next-polyfill-nomodule/src/index.js) so you likely have to polyfill it yourself via something like [this example](https://github.com/zeit/next.js/tree/canary/examples/with-polyfills)

**Comment by timneutkens** [comment]
We can probably include it! Feel free to send a PR.

**Comment by jamesmosier** [comment]
Actually there's no such thing as `Object.repeat` 😂 I forgot that is one of those obscure & misleading IE11 errors. It'll take some more digging to see why nanoid is not working in IE11. I looked at the transpiled output and nothing sticks out. The only catch is this module uses the browsers built-in `crypto` methods, so they want you to "Add self.crypto = self.msCrypto before Nano ID to fix IE 11 support" and I think that is the issue. @elliottsj

**Comment by elliottsj** [comment]
`window.crypto = window.msCrypto` did the trick; thanks @jamesmosier! I need to read docs more closely 😓

**Comment by elliottsj** [comment]
I may be looking in the wrong place, but it appears that [Next.js already polyfills `String.prototype.repeat`](https://github.com/zeit/next.js/blob/7d2ff81207d4d7ad6046a02fb171c2f38ea2fc34/packages/next-polyfill-nomodule/src/index.js#L38).

The error occurs when Next.js renders react-error-overlay; so seems that the polyfill isn't being applied prior to that point. I'll do some more digging...

**Comment by elliottsj** [comment]
Narrowed down the issue some more:

* The error `"TypeError: Object doesn't support property or method 'repeat'"` occurs on the expression `l(a.marker,"^").repeat(h)` in the minified source file `node_modules/next/dist/compiled/react-error-overlay/index.js`.
* The error is only produced when we throw an error at module execution time (rather than at render time).

    e.g. this is ok

    ```tsx
    export default () => {
      throw new Error('this is a test');
      return <span>hello world</span>;
    };
    ```

    but this fails

    ```tsx
    throw new Error('this is a test');
    export default () => (<span>hello world</span>);
    ```

Repro:

```shell
$ yarn create next-app my-app
$ cd my-app
$ echo "throw new Error('this is a test'); export default () => (<span>hello world</span>);" > pages/index.js
$ yarn dev
```

Open the app in IE11, and you'll see the error:

![IE11 JS console error](https://user-images.githubusercontent.com/2177366/80007756-b8486580-8494-11ea-8dd0-c842336d4311.png)

Observe that `String.prototype.repeat` is undefined when the `l(a.marker,"^").repeat(h)` expression is evaluated by replacing it with

```js
(function(){console.log("String.prototype.repeat",String.prototype.repeat);return l(a.marker,"^").repeat(h);})()
```

e.g.

```shell
$ sed -i '' 's/l(a.marker,"^").repeat(h)/(function(){console.log("String.prototype.repeat",String.prototype.repeat);return l(a.marker,"^").repeat(h);})()/g' node_modules/next/dist/compiled/react-error-overlay/index.js
$ yarn dev
```

<img width="801" alt="Annotation 2020-04-28 120248" src="https://user-images.githubusercontent.com/2177366/80511366-01dbf900-894a-11ea-9442-18d41f2ea2a9.png">

So, the issue seems to be that [Next.js's polyfills module](https://github.com/zeit/next.js/blob/7d2ff81207d4d7ad6046a02fb171c2f38ea2fc34/packages/next-polyfill-nomodule/src/index.js) is not being loaded prior to executing the `pages/index.js` module.

**Comment by mattdell** [comment]
I'm having this problem. Was there a solution or fix for this @elliottsj?

**Comment by elliottsj** [comment]
@mattdell There's no fix yet, afaik. The above repro steps should help someone get started with implementing a fix, but I'm not continuing this investigation myself at the moment.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
